packages/botonic-plugin-flow-builder/src/api.ts

```diff
@@ -0,0 +1,162 @@
+    }
+    const fallbackFirstMessage = fallbackNode.content.first_message
+    if (!fallbackFirstMessage) {
+      throw new Error('Fallback 1st message must be defined')
+    }
+    const fallbackSecondMessage = fallbackNode.content.second_message
+    if (!fallbackSecondMessage) {
+      return this.getNode(fallbackFirstMessage.id)
+    }
+    return alternate
+      ? this.getNode(fallbackFirstMessage.id)
+      : this.getNode(fallbackSecondMessage.id)
+  }
+
+  getNodeByIntent(input: Input, locale: string): HtNodeWithContent | undefined {
+    try {
+      const intents = this.flow.nodes.filter(
+        node => node.type === HtNodeWithContentType.INTENT
+      ) as HtIntentNode[]
+      const inputIntent = input.intent
+      const inputConfidence = input.confidence
+      if (inputIntent) {
+        const matchedIntentNode = intents.find(
+          node =>
+            inputIntent &&
+            this.hasIntent(node, inputIntent, locale) &&
+            inputConfidence &&
+            this.hasMetConfidenceThreshold(node, inputConfidence)
+        )
+        return (
+          matchedIntentNode?.target &&
+          this.getNode<HtNodeWithContent>(matchedIntentNode?.target.id)
+        )
+      }
+    } catch (error) {
+      console.error(`Error getting node by intent '${input.intent}': `, error)
+    }
+
+    return undefined
+  }
+
+  private hasIntent(
+    node: HtIntentNode,
+    intent: string,
+    locale: string
+  ): boolean {
+    return node.content.intents.some(
+      i => i.locale === locale && i.values.includes(intent)
+    )
+  }
+
+  private hasMetConfidenceThreshold(
+    node: HtIntentNode,
+    predictedConfidence: number
+  ): boolean {
+    const nodeConfidence = node.content.confidence / 100
+    return predictedConfidence >= nodeConfidence
+  }
+
+  getNodeByKeyword(
+    userInput: string,
+    locale: string
+  ): HtNodeWithContent | undefined {
+    try {
+      const keywordNodes = this.flow.nodes.filter(
+        node => node.type == HtNodeWithContentType.KEYWORD
+      ) as HtKeywordNode[]
+      const matchedKeywordNodes = keywordNodes.filter(node =>
+        this.matchKeywords(node, userInput, locale)
+      )
+      if (matchedKeywordNodes.length > 0 && matchedKeywordNodes[0].target) {
+        return this.getNode<HtNodeWithContent>(matchedKeywordNodes[0].target.id)
+      }
+    } catch (error) {
+      console.error(`Error getting node by keyword '${userInput}': `, error)
+    }
+
+    return undefined
+  }
+
+  private matchKeywords(
+    node: HtKeywordNode,
+    input: string,
+    locale: string
+  ): boolean {
+    const result = node.content.keywords.find(
+      i => i.locale === locale && this.containsAnyKeywords(input, i.values)
+    )
+    return Boolean(result)
+  }
+
+  private containsAnyKeywords(input: string, keywords: string[]): boolean {
+    for (let i = 0; i < keywords.length; i++) {
+      if (input.includes(keywords[i])) {
+        return true
+      }
+    }
+    return false
```

Comment:
This can be simplified to` return !!keywords.find((keyword) => input.includes(keyword))`

Reply:
Done!